### ISSUE TITLE
New build is not Windows friendly

### ISSUE THREAD

**Issue by chunghha** [vmware/clarity#1582]
**Select one ...**  (check one with "x")
```
[ ] bug
[ ] feature request
[x] enhancement
```

### Expected behavior

"npm run build" should be error-free on Windows.
### Actual behavior

"npm run build" fails on Windows. "del ./dist" and the cpy npm tasks are troulbesome on Windows. Running on Linux is fine though.

### Reproduction of behavior

* Include a link to the reproduction scenario you created by forking one of the Clarity Plunker Templates:

### Environment details

* **Angular version:** 4.x.x

* **Clarity version:** master https://github.com/vmware/clarity/commit/eca43905caf18cdcec3f0a81d61fd370051d9c80

* **OS and version:** 
Windows 7

* **Browser:** [all | Chrome XX | Firefox XX | IE XX | Safari XX | Mobile Chrome XX | Android X.X Web Browser | iOS XX Safari | iOS XX UIWebView | iOS XX WKWebView ]

**Comment by adityarb88** [comment]
@jeeyun Could you please take a look at this?

**Comment by hippee-lee** [comment]
I'll take a look at it.

**Comment by hippee-lee** [comment]
Turns out `del` and `cpy` don’t play nice with windows

https://github.com/sindresorhus/cpy-cli/issues/9

I can get `del-cli` to work but nothing doing for cpy. If we need to support windows (I think yeah, even though I don’t want to) it seems like we will need to change the delete/copy tools commands for our build. From the author:
``` I couldn't care less about Windows, but feel free to change it if you do. ```

**Comment by chunghha** [comment]
How about using [shx](https://github.com/shelljs/shx)?

**Comment by hippee-lee** [comment]
@chunghha - that looks interesting. Thanks!
I'll take a look and see it it works a bit more seamlessly for *nix && Windows.

**Comment by hippee-lee** [comment]
Ran into issues with many things on my windows VM. The related thing to his issue is that the `replace` command we are using seems have a problem with `$npm_package_version - it seems windows wants %npm_package_version% but I can't confirm that for sure as I'm getting an error that is quite troublsome to debug: `cp: copyFileSync: could not write to dest file (code=ENOENT)` . 

A possible solution might be to use [crossvar](https://www.npmjs.com/package/cross-var)

I'm going to move this to the backlog for now and mark it as contributions welcome for someone who has access to an ideal Windows dev setup for node.

**Comment by chunghha** [comment]
So far this still works on Linux, will test on Windows. 
https://gist.github.com/anonymous/99cf282d9573d9ec423a0be260896703

**Comment by chunghha** [comment]
The above error is from Windows 10/node 8.7.0. But, from Windows7/node 6.11.4, the above error disappears. Just mention that here for record now. Since I tested with node 8.7.0 on my Linux, I am not sure node version is related to the error during webpack on Windows 10.

With Windows7/node 6.11.4, one error I am getting is from this code in clarify-icons-svg.js
```
SHAPE_SETS.forEach((setName) => {
        shell.exec(`cd dist/clarity-icons/shapes; zip -r ${setName}.zip ./${setName}/*; rm -r ./${setName}`);
    });
```
The error is "The system cannot find the path specified." 7 times so taking care of the cd part, then error say 

> 'zip' is not recognized as an internal or external command

so basically, need to re-write the clarify-icon-svg.js to make it seamlessly work on Windows.

**Comment by chunghha** [comment]
@hippee-lee when you have time, please take a look at this.
https://gist.github.com/chunghha/5cc00f1021d2aaf26277ce2667c4095e
This is a working version on my Windows7/node 6.11.4 along with the shx/cross-conf-env/replace-in-file changes above. I haven't fully verified contents of all-shapes.zip are matched with the one from Linux but svg icons in other shape zip files looked OK.

When I can get a chance (probably this weekend), I will create a branch on my fork repo and do more tests with Windows 7/10 and Linux.

**Comment by hippee-lee** [comment]
Thank you for continuing to look into this! What I see in your gist looks like a viable approach. 

If you want, feel free to submit a PR with your suggested changes. Then, I can pull it on my end and test it out my Windows VM.

**Comment by vinayakpatil** [comment]
Will this be merged anytime soon?

**Comment by hippee-lee** [comment]
Hi @vinayakpatil -

I'm really sorry but I haven't had time to come back to this. I'll update the PR this issue is linked to with feedback this week.

**Comment by sosharma1403** [comment]
Any updates on this?

**Comment by gnomeontherun** [comment]
Actually, we just merged a revamp of our build tooling. It uses the Angular CLI for most of the work so it might do better on Windows. Would you mind testing against master and if you have issues report back?

**Comment by mathisscott** [comment]
I'm assuming that the new build tooling addressed this.
